proofs/erc20.md

```diff
@@ -0,0 +1,1030 @@
+        <callDepth> CALL_DEPTH </callDepth>
+        <callStack> _ => _ </callStack>
+        <interimStates> _ </interimStates>
+        <substateStack> _ </substateStack>
+        <callLog> .Set </callLog> // for vmtest only
+
+        <txExecState>
+          <program> #asMapOpCodes(#dasmOpCodes(#parseByteStack({code}), DEFAULT)) </program>
+          <programBytes> #parseByteStack({code}) </programBytes>
+
+          <id> ACCT_ID </id> // contract owner
+          <caller> CALLER_ID </caller> // who called this contract; in the begining, origin
+
+          <callData> {calldata} </callData>
+
+          <callValue> 0 </callValue>
+          <wordStack> .WordStack => _ </wordStack>
+          <localMem>
+            {localmem}
+          </localMem>
+          <pc> 0 => _ </pc>
+          <gas> {gas} </gas>
+          <previousGas> _ => _ </previousGas>
+
+          <static> false </static> // NOTE: non-static call
+        </txExecState>
+
+        <substate>
+          <selfDestruct> _ </selfDestruct>
+          <log>
+            {log}
+          </log>
+          <refund> {refund} </refund> // TODO: more detail
+        </substate>
+
+        <gasPrice> _ </gasPrice>
+        <origin> ORIGIN_ID </origin> // who fires tx
+
+        <previousHash> _ </previousHash>
+        <ommersHash> _ </ommersHash>
+        <coinbase> _ </coinbase>
+        <stateRoot> _ </stateRoot>
+        <transactionsRoot> _ </transactionsRoot>
+        <receiptsRoot> _ </receiptsRoot>
+        <logsBloom> _ </logsBloom>
+        <difficulty> _ </difficulty>
+        <number> _ </number>
+        <gasLimit> _ </gasLimit>
+        <gasUsed> _ </gasUsed>
+        <timestamp> _ </timestamp>
+        <extraData> _ </extraData>
+        <mixHash> _ </mixHash>
+        <blockNonce> _ </blockNonce>
+
+        <ommerBlockHeaders> _ </ommerBlockHeaders>
+        <blockhash> _ </blockhash>
+      </evm>
+
+      <network>
+        <activeAccounts> ACCT_ID |-> false _:Map </activeAccounts>
+
+        <accounts>
+          <account>
+            <acctID> ACCT_ID </acctID>
+            <balance> _ </balance>
+            <code> #parseByteStack({code}) </code>
+            <storage>
+              {storage}
+            </storage>
+            <nonce> _ </nonce>
+          </account>
+       // ... // TODO: fix
+        </accounts>
+
+        <txOrder> _ </txOrder>
+        <txPending> _ </txPending>
+        <messages> _ </messages>
+      </network>
+    </ethereum>
+    requires 0 <=Int ACCT_ID    andBool ACCT_ID    <Int (2 ^Int 160)
+     andBool 0 <=Int CALLER_ID  andBool CALLER_ID  <Int (2 ^Int 160)
+     andBool 0 <=Int ORIGIN_ID  andBool ORIGIN_ID  <Int (2 ^Int 160)
+     andBool 0 <=Int CALL_DEPTH andBool CALL_DEPTH <Int 1024
+     {requires}
+
+{comments}
+```
+
+Below are an abstract semantics of the four instruction, LT, GT, EQ, and ISZERO, that improves the performance of the underlying theorem prover’s symbolic reasoning by avoiding unnecessary case analyses.
```

Comment:
shouldn't these go somewhere else, where we have lemmas and axioms?

Reply:
I also want to do that, but I can't due to the limitation of the current k prover. But, we have an outstanding PR related to this, so hopefully we can improve that soon.

Reply:
cool.  for now, add this as a comment; then we change it when we fix it